src/js/controller/LayersListController.js

```diff
@@ -130,6 +133,18 @@
       layerNameEl.setAttribute('title', layer.getName());
       layerNameEl.setAttribute('rel', 'tooltip');
     }
+    if (isRenaming) {
+      var layerNameInputHtml = pskl.utils.Template.replace(this.layerNameInputTemplate_, {
+        'layername': layer.getName()
+      });
+      var layerNameInput = pskl.utils.Template.createFromHTML(layerNameInputHtml);
+      var layerNameEl = layerItem.querySelector('.layer-name');
+      layerItem.replaceChild(layerNameInput, layerNameEl);
+      layerNameInput.addEventListener('blur', this.onRenameCurrentyLayerDone_.bind(this));
```

Comment:
Here we will be leaking these two event listeners every time we create this input. In `ns.LayersListController.prototype.init` it is ok to do it because we will only call init() once in the whole lifecycle of the application. 

Here the method is called every time we create the input so we have to be more careful.
You can either use event delegation or remove the events when setting `this.isRenaming` to `false`. I think the second approach will be easier so a few pointers:
- bind the method once and for all in the constructor: `this.onRenameCurrentyLayerDone_ = this.onRenameCurrentyLayerDone_.bind(this)`
- attach your listeners without calling bind() again `layerNameInput.addEventListener('blur', this.onRenameCurrentyLayerDone_)`
- remove the event listeners right before setting this.isRenaming to false
- to be safe, also remove the event listeners right before adding them (this way if addLayerItem is called twice for some reason, the events will only be added once)

Reply:
Very nice feedback, thanks. Done.